### PR TITLE
[PageBuilder] Drag & drop item sorting issues

### PR DESCRIPTION
**Pull request by aachavarria**
Content.tsx: using performMutation on SortItem;
ContentController.tsx: removing no used insertItem();
ContentController.tsx: moving getCollection to a fn;
ContentController.tsx: changes on sortItem to supportNested sort;
Guest.tsx: Fixing issue on with highlight after sort;
Guest.tsx: sending the correct index to sortItem();
GuestProxy.tsx: issue on MoveItem, when sent a item to the last of the list

### CODE REVIEW

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 1]
```diff
@@ -201,10 +207,6 @@ export function PreviewConcierge(props: any) {
             }
           );
           break;
-        case INSERT_ITEM_OPERATION: {
-          enqueueSnackbar('Insert item operation not implemented.');
-          break;
-        }
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: Why remove this?

**Review by rart — CHANGES_REQUESTED** [review]
Why is the insert op getting removed? 
This is meant to insert repeat group items; it hasn't been implemented but it's in the roadmap. What's your idea in removing them?

File: ui/app/src/modules/Preview/PreviewConcierge.tsx (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/modules/Preview/PreviewConcierge.tsx]: It says Insert item, not insert repeat group items, so I thought it was used to insert an items as insert instance did

Comment (aachavarria) [reply ui/app/src/modules/Preview/PreviewConcierge.tsx]: I added it again

File: ui/app/src/services/content.ts [hunk 2]
```diff
@@ -1174,7 +1153,6 @@ export default {
   updateField,
   insertComponent,
   insertInstance,
-  insertItem,
```
Comment (rart) [inline ui/app/src/services/content.ts]: 👆🏻

File: ui/guest/src/classes/ContentController.ts [hunk 3]
```diff
@@ -333,9 +333,6 @@ export class ContentController {
 
   }
 
-  insertGroup(modelId, fieldId, data): void {
-  }
-
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts]: 👆🏻

File: ui/guest/src/classes/ContentController.ts [hunk 4]
```diff
@@ -958,19 +961,21 @@ function reducer(lookupTable: LookupTable<ContentInstance>, model: ContentInstan
 
 }
 
-function getResult(model: ContentInstance, fieldId: string, index: string | number): number[] | string[] {
-  const isStringIndex = typeof index === 'string';
+//function getResult(model: ContentInstance, fieldId: string, index: string | number): number[] | string[] {
+function getResult(collection: string[], index): string[] {
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts]: This looks temporary. What's the idea here?

File: ui/guest/src/classes/ContentController.ts [hunk 5]
```diff
@@ -958,19 +961,21 @@ function reducer(lookupTable: LookupTable<ContentInstance>, model: ContentInstan
...
   const parsedIndex = parseInt(popPiece(`${index}`), 10);
-
-  const collection = isStringIndex
-    ? ModelHelper.extractCollection(model, fieldId, index)
-    : ModelHelper.value(model, fieldId);
-
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts:968]: This looks dangerous. Sure? Need to test all kinds of sort/insert

**Review by rart — CHANGES_REQUESTED** [review]
Review/resolve above comments.

File: ui/guest/src/classes/ContentController.ts (hunk 5 above)
Comment (aachavarria) [reply ui/guest/src/classes/ContentController.ts:968]: yes its works very well

File: ui/guest/src/classes/ContentController.ts (hunk 4 above)
Comment (aachavarria) [reply ui/guest/src/classes/ContentController.ts]: sometimes we need to collection(insert item), but others we need the result( collection - the sort item)

File: ui/guest/src/classes/ContentController.ts [hunk 6]
```diff
@@ -958,19 +964,20 @@ function reducer(lookupTable: LookupTable<ContentInstance>, model: ContentInstan
 
 }
 
-function getResult(model: ContentInstance, fieldId: string, index: string | number): number[] | string[] {
-  const isStringIndex = typeof index === 'string';
+function getResult(collection: string[], index: string | number): string[] {
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts]: Rename to `getCollectionWithoutItemAtIndex`

**Review by rart — APPROVED** [review]

**Pull request merged**